pkg/api/types.go

```diff
@@ -1179,11 +1215,22 @@ type BundleSourceStepConfiguration struct {
 // PipelineImageStreamTagReferenceBundleSourceName is the name of the bundle source image built by the CI
 const PipelineImageStreamTagReferenceBundleSource PipelineImageStreamTagReference = "src-bundle"
 
-// BundlePrefix is the prefix used by ci-operator for bundle images
+// BundlePrefix is the prefix used by ci-operator for bundle images without an explicitly configured name
```

Comment:
Do we have validation that ensures this name does not collide with any `base_images` or `images` or baseline `pipeline` images?

Reply:
I had some new validation. I have updated this PR to add a lot more validation. It should now detect any collisions between bundle+index names and base and built images.

Something I noticed is that we don't seem to validate for collisions between built images and base images. Should we add validation for that (in a separate PR)?

Reply:
Wow, yeah we should, since they both target `pipeline:`